File: iOS4bookExamples/p707p723map/p707map/MyAnnotation.m


#import "MyAnnotation.h"


@implementation MyAnnotation

@synthesize coordinate, title, subtitle;

- (void)dealloc {
    [title release];
    [subtitle release];
    [super dealloc];
}

- (id)initWithLocation: (CLLocationCoordinate2D) coord {
    self = [super init];
    if (self) {
        self->coordinate = coord;
    }
    return self;
}

@end

File: iOS9bookExamples/bk2ch21p744maps/ch34p1008maps/MyAnnotation.swift
import UIKit
import MapKit

class MyAnnotation : NSObject, MKAnnotation {
    dynamic var coordinate : CLLocationCoordinate2D
    var title: String?
    var subtitle: String?
    
    init(location coord:CLLocationCoordinate2D) {
        self.coordinate = coord
        super.init()
    }

}
